### DISCUSSION TITLE
Serving (previous) generated page instead of fallback after new build

### DISCUSSION THREAD

**Discussion by tbgse** [vercel/next.js#17417] [Ideas]
We are deploying quite regularly daily, which means our pages that have previously been generated on the fly will now all show the fallback again, instead of immediately loading from the CDN  (which is obvious, since the pages might have changed after a new deployment so they will require to be regenerated).

We really would love to reduce the amount of instances where users are seeing fallbacks, but still continue to take advantage of site generation during runtime. We were wondering if there would be a way to instead of the fallback serve the last valid (stale) version of a page if one has already been generated for a previous release and generate the new version in the background so that the updated version would be served afterwards for all users. This would have the downside that some users might be seeing a stale page after a new update was deployed, but the upside is that no users would ever have to see a fallback page again, no matter how often you deploy. With critical deployments there could be an option to invalidate the cache and force all pages to render with a fallback again.

Really curious if this is something that could be viable.

**Comment by adduss** [comment]
@tbgse did you managed to implement this behaviour in any way (for example using reverse proxy or redirects)?

**Comment by tbgse** [comment]
Unfortunately not, we now have a warmup script that crawls all pages to
trigger revalidation after a new build was done. It's not as nice but it
works ok

**Reply by adduss** [reply]
thank you @tbgse for quick response! Unfortunately we are aiming to have 6M+ statically generated pages and such warmup would take couple of days :/
